src/vcpkg/base/downloads.cpp

```diff
@@ -988,10 +997,22 @@ namespace vcpkg
                 }
             }
         }
-        msg::println_error(msgFailedToDownloadFromMirrorSet);
+        // Asset cache is not configured and x-block-origin enabled
+        if (m_config.m_read_url_template.has_value())
+        {
+            msg::println(msgAssetCacheMissBlockOrigin, msg::path = download_path.filename());
+        }
+        else
+        {
+            msg::println_error(msgMissingAssetBlockOrigin, msg::path = download_path.filename());
+        }
+
         for (LocalizedString& error : errors)
         {
-            msg::println(error);
+            // Debug only since:
+            // 1. curl already prints the error
+            // 2. the printed url can contain sensitive information like username/password
+            Debug::println(error);
```

Comment:
I know you said this change was intentional over chat but I don't really agree with it. Printing secrets is unfortunate but making it impossible for users to debug the problem because we don't show them the information they need is worse.

Reply:
Not impossible. They would need to use `--debug` but I get that it's not localized and not meant to be user facing.